src/utils/video.ts

```diff
@@ -10,3 +10,41 @@ export const formatVideoViewsAndDate = (
   const formattedViews = views !== null && (fullViews ? formatNumber(views) : formatNumberShort(views))
   return formattedViews ? `${formattedViews} views • ${formattedDate}` : formattedDate
 }
+
+export type VideoMetadata = {
+  mimeType: string
+  sizeInBytes: number
+  height: number
+  width: number
+  duration: number
+}
+export const getVideoMetadata = async (
+  file?: File
+): Promise<{ metadata: VideoMetadata; error?: undefined } | { metadata?: undefined; error: Error }> => {
+  const videoEl = document.createElement('video')
+  const mimeType = file?.type ?? ''
+  const sizeInBytes = file?.size ?? 0
+  const Url = URL
+  const canPlay = videoEl.canPlayType(mimeType)
+
+  return new Promise((resolve) => {
+    const handleLoadeddata = (e: Event) => {
+      // Video should now be loaded but we can add a second check
+      if (videoEl.readyState >= 3) {
```

Comment:
If it's in some other state, let's reject the promise, because I guess otherwise it will be pending forever?